frontend/src/app/[url_key]/page.tsx

```diff
@@ -0,0 +1,80 @@
+'use client';
+
+import { ApiClient } from "@/utils/api";
+import { useEffect, useState } from "react";
+import { useRouter } from "next/navigation";
+import { ApiClientError } from "@/utils/api";
+
+/**
+ * Displays pages from page management system.
+ *
+ * Each page is identified with a unique url_key. When visiting /[url_key], the page with that url_key is fetched
+ * from the backend if it exists.
+ */
+export default function Page({ params }: { params: any }) {
+    const router = useRouter();
+    const { url_key } = params;
+
+    const [page, setPage] = useState<any>(null);
+    const [error, setError] = useState<ApiClientError | null>(null);
+
+    useEffect(() => {
+        const FetchData = async () => {
+            try {
+                const res = await ApiClient('GET', `/api/pages/${url_key}`);
+
+                if (res.error) {
+                    setError({ message: res.error.message, status: res.error.code });
```

Comment:
Backend gives inconsistent errors. On 401 it is message and code, on 404 it is detail and status. It should all be detail and status. Ticket is created. This error should be res.error.status and res.error.detail

Reply:
https://burgieclan.atlassian.net/browse/BUR-170